### DISCUSSION TITLE
Version 2.11 injects proxy containers at the top instead of at the bottom

### DISCUSSION THREAD

**Discussion by EraYaN** [linkerd/linkerd2#7062] [Q&A] [answered]
So 2.11 seems to (auto) inject proxy containers at the top of the manifest instead of the bottom (as it did with 2.10).

This means that for a many tasks (kubectl exec, logs etc) the linker-proxy is now the default container and not the application container.
Is there a configuration item or otherwise design choice on why this was moved to the front?

**Comment by alpeb** [comment]
The reasoning behind this change was so that the application container only starts after the proxy has been started and is ready. This fixes situations where the app might hit the network too early and would fail because the connection wasn't being proxied. You can read more details about this change in #5967. There is no setting available to modify this behavior.

**Reply by wmorgan** [reply]
It's a bummer that this doesn't play nicely with kubectl commands. I'm not sure what to do here -- this change fixes a real issue for many users.

**Comment by kleimkuhler** [comment, marked as answer]
@alpeb summarized the reason for this change well so let us know if you have any more questions about the _why_.

This can be disabled though. If you want the proxy to be added after all existing containers—as it did with 2.10—you can disable it with the `config.linkerd.io/proxy-await: "disabled"`. Keep in mind this can lead to the issues with container startup ordering.

**Reply by EraYaN** [reply]
Our services are tolerant to the linkerd-proxy startup time so we can just disable it.

**Reply by kleimkuhler** [reply]
Great!
